EAO/UI/Inspections/InspectionFormController.swift

```diff
@@ -43,27 +53,29 @@ final class InspectionFormController: UIViewController{
         addButton.layer.shadowRadius = 4
     }
 
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        
+        if segue.identifier == InspectionFormController.showInspectionFormSegueID, let destinationVC = segue.destination as? InspectionSetupController {
+            destinationVC.inspection = inspection
+        }
+        
+        if segue.identifier == InspectionFormController.addNewElementSegueID, let destinationVC = segue.destination as? NewObservationElementFormViewController {
+            destinationVC.observation = nil
+            destinationVC.inspection = inspection
+        }
+        
+        if segue.identifier == InspectionFormController.editElementSegueID, let destinationVC = segue.destination as? NewObservationElementFormViewController, let index = tableView.indexPathForSelectedRow?.row, index < observations.count {
+            destinationVC.inspection = inspection
+            destinationVC.observation = observations[index]
+            destinationVC.isReadOnly = isReadOnly
+        }
+    }
+    
 	@IBAction fileprivate func backTapped(_ sender: UIBarButtonItem) {
 		sender.isEnabled = false
-		pop()
+		popViewController()
```

Comment:
In this an all other cases below all the information is either sent back via callbacks or saved into realm before transitions. And since unwind segue is going to be called from code, I think that using it adds extra code and complexity without any benefits